src/components/dount/utils.tsx

```diff
@@ -0,0 +1,121 @@
+import * as d3 from "d3";
+import { createPopper, Instance, Instance as PopperInstance } from '@popperjs/core';
+import './styles.scss';
+import _ from 'lodash';
+import { colors } from "./chartdata";
+import { ModifiedData } from "./types";
+
+const DrawDonut = (element: HTMLElement, data: ModifiedData[]) => {
+  const width = 600;
+  const height = 400;
+  const innerRadius = 100;
+  const outerRadius = Math.min(width, height) / 2;
+  const radius = Math.min(width, height) / 1.3;
+  const innerHole = radius / 3.3;
+  const legendPosition = d3.arc().innerRadius(radius / 1.75).outerRadius(radius);
+  const totalShare = `${_.sumBy(data, d => parseInt(d.sharePercent))}%`
+  d3.select(element).select("svg").remove();
+
+  const svg = d3.select(element)
+    .append('svg')
+    .attr("width", '400px')
+    .attr("height", '420px')
+    .attr('viewBox', '0 0 ' + width + ' ' + height)
+    .append("g")
+    .attr("transform", "translate(" + Math.min(width, height) / 1.5 + "," + Math.min(width, height) / 1.8 + ")");
+
+  const pieGenerator = d3.pie<ModifiedData>()
+    .value(d => d.value);
+  const arcGenerator = pieGenerator(data);
+  const arc = d3.arc<d3.PieArcDatum<ModifiedData>>().innerRadius(innerRadius).outerRadius(outerRadius - 30);
+  const tooltipInstances: Map<HTMLElement, Instance> = new Map();
+
+  svg
+    .selectAll('g')
```

Comment:
values should be formatted to bignumber and make the values default around the circle instead of showing percentage and tooltip should show the true value